rskj-core/src/test/java/co/rsk/peg/BridgeSupportTest.java

```diff
@@ -2925,6 +2925,182 @@ public void addLockWhitelistAddress_invalidAddress() throws IOException {
+        byte[] senderBytes = ECKey.fromPublicOnly(Hex.decode(
+            // Public key hex of the authorized whitelist admin in regtest, taken from BridgeRegTestConstants
+            "04641fb250d7ca7a1cb4f530588e978013038ec4294d084d248869dd54d98873e45c61d00ceeaeeb9e35eab19fa5fbd8f07cb8a5f0ddba26b4d4b18349c09199ad"
+        )).getAddress();
+        RskAddress sender = new RskAddress(senderBytes);
+        when(mockedTx.getSender()).thenReturn(sender);
+        LockWhitelist mockedWhitelist = mock(LockWhitelist.class);
+        when(mockedWhitelist.isDisableBlockSet()).thenReturn(false);
+        int bestChainHeight = (Integer.MAX_VALUE / 2) + 2;
+        BtcBlockstoreWithCache btcBlockStore = mock(BtcBlockstoreWithCache.class);
+        StoredBlock storedBlock = mock(StoredBlock.class);
+        when(storedBlock.getHeight()).thenReturn(bestChainHeight);
+        when(btcBlockStore.getChainHead()).thenReturn(storedBlock);
+        BridgeSupport bridgeSupport = getBridgeSupportWithMocksAndBtcBlockstoreForWhitelistTests(mockedWhitelist,btcBlockStore);
+
+        BigInteger disableBlockDelayBI = BigInteger.valueOf(Integer.MAX_VALUE / 2);
+
+        Assert.assertEquals(-2, bridgeSupport.setLockWhitelistDisableBlockDelay(mockedTx, disableBlockDelayBI).intValue());
+        verify(mockedWhitelist, never()).put(any(), any());
+    }
+
+    @Test
+    public void setLockWhitelistDisableBlockDelay_maxValueDisableBlockBI() throws IOException, BlockStoreException {
+        Transaction mockedTx = mock(Transaction.class);
+        byte[] senderBytes = ECKey.fromPublicOnly(Hex.decode(
+            // Public key hex of the authorized whitelist admin in regtest, taken from BridgeRegTestConstants
+            "04641fb250d7ca7a1cb4f530588e978013038ec4294d084d248869dd54d98873e45c61d00ceeaeeb9e35eab19fa5fbd8f07cb8a5f0ddba26b4d4b18349c09199ad"
+        )).getAddress();
+        RskAddress sender = new RskAddress(senderBytes);
+        when(mockedTx.getSender()).thenReturn(sender);
+        LockWhitelist mockedWhitelist = mock(LockWhitelist.class);
+        when(mockedWhitelist.isDisableBlockSet()).thenReturn(false);
+        int bestChainHeight = 10;
+        BtcBlockstoreWithCache btcBlockStore = mock(BtcBlockstoreWithCache.class);
+        StoredBlock storedBlock = mock(StoredBlock.class);
+        when(storedBlock.getHeight()).thenReturn(bestChainHeight);
+        when(btcBlockStore.getChainHead()).thenReturn(storedBlock);
+        BridgeSupport bridgeSupport = getBridgeSupportWithMocksAndBtcBlockstoreForWhitelistTests(mockedWhitelist,btcBlockStore);
+
+        BigInteger disableBlockDelayBI = BigInteger.valueOf(Integer.MAX_VALUE);
+
+        Assert.assertEquals(-2, bridgeSupport.setLockWhitelistDisableBlockDelay(mockedTx, disableBlockDelayBI).intValue());
+        verify(mockedWhitelist, never()).put(any(), any());
+    }
+
+    @Test
+    public void setLockWhitelistDisableBlockDelay_negativeBestChain() throws IOException, BlockStoreException {
```

Comment:
I don't think we need to cover this case.

Reply:
good catch